app/gui2/src/components/GraphEditor/widgets/WidgetText.vue

```diff
@@ -60,9 +89,19 @@ export const widgetDefinition = defineWidget(WidgetInput.isAstOrPlaceholder, {
 </script>
 
 <template>
-  <label class="WidgetText r-24" @pointerdown.stop>
+  <label ref="widgetRoot" class="WidgetText r-24" @pointerdown.stop>
     <NodeWidget v-if="shownLiteral.open" :input="WidgetInput.FromAst(shownLiteral.open)" />
-    <AutoSizedInput v-model.lazy="textContents" />
+    <AutoSizedInput
+      ref="input"
+      v-model="editedContents"
+      autoSelect
+      @pointerdown.stop
+      @pointerup.stop
+      @click.stop
+      @keydown.enter.stop="accepted"
```

Comment:
Not sure, but `escape` can probably cancel?

Reply:
It cancels already, but through our interaction system. Each active interaction is cancelled by Esc (caught in capture phase on window level).